### DISCUSSION TITLE
Keys persistence

### DISCUSSION THREAD

**Discussion by KeyMag** [xtrime-ru/antizapret-vpn-docker#6] [Q&A] [answered]
Как при обновлении контейнера сохранить сертификаты и файлы конфигураций. Что бы по новой не распространять на клиентские устройства?

**Comment by xtrime-ru** [comment]
По умолчанию все важные данные хранятся на хосте в смонтированных папках  easyrsa3/pki/ и client_keys/. Рестартовать и пересоздавать контейнер - безопасно.

Соотсветственно, что бы пересоздать ключи, нужно удалить эти папки: 
https://github.com/xtrime-ru/antizapret-vpn-docker?tab=readme-ov-file#keys-menagment

**Reply by KeyMag** [reply]
Тогда как правильно обновить контейнер?
`git fetch && git reset --hard origin/master docker compose build docker compose up -d`
Данные команды обновили все:)

**Comment by KeyMag** [comment]
И так через `git pull` подтянул актуальный релиз, через `docker compose up -d --build` обновил контейнер, добавление проблемного сайта в exclude-hosts-custom.txt проблему не решает. Но если сайт добавить в include-hosts-custom.txt то через туннель данный сайт работает и работает если не используется туннель. Как поправить такое поведение? Сайт myspar.ru, такая же проблема с сайтом chiphell.com

**Reply by xtrime-ru** [reply, marked as answer]
Возможно проблема в том, что сайт блокирует запросы, так как dns запрос был из другой локации. Попробуй добавить домен с точкой на конце в этот список и заново сбилдить контейнер: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/Dockerfile#L32

**Reply by KeyMag** [reply]
Внесение сайтов помогло, спасибо. 
`        policy.todnames({'apple.com.', 'mzstatic.com.', 'akamaiedge.net.', 'edgekey.net.', 'aaplimg.com.', 'chiphell.com.', 'myspar.ru.'})\n\`
Можно ли этого избежать как ни будь? Просто при использовании оригинального решения всё работает.

**Reply by xtrime-ru** [reply]
Возможно, что в общедоступном варианте автор точно так же внес руками этот домен в аналогичный список. Либо общедоступный вариант хостится в РФ или использует российский DNS. Суть исправления заключается в том, что бы запрашивать эти домены в российском ДНС, а не в google/cloudflare. Соответственно запрос в DNS и запрос к сайту совершаются из единой геозоны и защита сайта не триггерится. 

Проблема не на нашей стороне, а это защита сайта не дает совершать запросы на адрес в другой гео зоны. На нашей стороне обойти это можно только такими правками, либо вручную внести в hosts файл на клиентах, что бы они не ходили в DNS за этим адресом.

**Reply by KeyMag** [reply]
Понял, спасибо! Буду мониторить данное поведение.

**Comment by KeyMag** [comment]
Столкнулся с такой проблемой есть определенный сайт который не заруливает в туннель при добавлении его include-hosts-custom.txt, пробовал его вносить в Dockerfile, тоже не помогло, nslookup возвращает IP сайта. Есть идеи что это может быть и куда копать?

**Comment by KeyMag** [comment]
Обнаружил что через данный VPN не прогружаются нормально мобильные карты Yandex, в том числе и пробки.

**Reply by xtrime-ru** [reply]
Возможно проблема в том, что DNS запрос идет из другой локации и поэтому их балансировщики не пропускают запрос из неправильного гео. 

Вот тут есть исключение, что бы некоторые домены шли как раз через DNS яндекса, что бы избежать этой проблемы: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/Dockerfile#L32

Можно попробовать добавить туда домены самого яндекса. Можно прям на работающем контейнере поправить конфиг и перезапустить контейнер или kres.d
